scripts/setup_iso.py

```diff
@@ -102,17 +114,44 @@ def determine_os_ver(srcdir, iso_info):
         elif 'ESXI-5.5' in vid:
             osver='5.5'
     else:
-        list=os.listdir(srcdir)
-        if "RPM-GPG-KEY-redhat-release" in list:
+        src_dir_list=os.listdir(srcdir)
+        if "RPM-GPG-KEY-redhat-release" in src_dir_list:
             osname = 'RHEL'
             osver = '7.0'
-        elif "RPM-GPG-KEY-CentOS-Testing-7" in list:
+        elif "RPM-GPG-KEY-CentOS-Testing-7" in src_dir_list:
             osname = "Centos"
             osver = '7.0'
-        elif 'LiveOS' and 'isolinux' in list:
+        elif 'isolinux' in src_dir_list:
             print 'attempting LiveCD netboot'
             osname = 'LiveCD'
             osver = vid
+        elif 'ubuntu' in src_dir_list:
+            # For ubuntu, the osver is combined by the version name(add -server for unbuntu server and without -server
+            # for desktop) and version number, e.g.ubuntu-server-16.04
+            osname='Ubuntu'
+            if vid == '': assert ValueError, "The volume ID is not get correctly"
+            ver_pattern = '(\w+\-?\w+)\s+(\d+\.?\d+\.?\d+?)'
+            m = re.search(ver_pattern, vid)
+            assert m.group(1), "The version name(e.g., unbuntu-server) for ubuntu is not correct"
+            assert m.group(2), "The version for ubuntu is not correct"
+            osver = m.group(1) + '-' + m.group(2)
+        elif 'suse' in src_dir_list:
+            osname='SUSE'
+            if vid == '': assert ValueError, "The volume ID is not get correctly"
+            ver_pattern = '(.+)-DVD-(.+)'
+            m = re.search(ver_pattern, vid)
+            assert m.group(1), "The version name is not retrieved correctly"
+            osver = m.group(1)
+        elif 'setup.exe' in src_dir_list:
+            osname, osver = get_setup_exe_name_version(srcdir)
+        else:
+             for i in src_dir_list:
+                 if "PHOTON" in i:
+                     osname="PHOTON"
+                     if vid == '': assert ValueError, "The volume ID is not get correctly"
+                     osver = vid.lstrip('PHOTON_')
```

Comment:
Thanks Lyne. The break is not need here since the Photon check is in the last else statement. In the future if more OS is added, we need add other if below Photon check